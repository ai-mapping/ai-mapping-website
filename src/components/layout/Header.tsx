import Link from 'next/link';
import Image from 'next/image';
import React from 'react';
import { siteConfig } from '@/config/site';

// Import the client component for interactive controls
import HeaderControls from './HeaderControls';
import { WhatsAppIcon, TelegramIcon } from '@/components/ui/icons';

// Define types for navigation items
type NavItem = {
  href: string;
  label: string;
};

// Define types for social links
type SocialLink = {
  href: string;
  icon: React.ReactNode;
  className: string;
};

// Main props interface (without callback handlers)
export type HeaderProps = {
  logo: {
    fullName: string;
    image: string;
    textImage: string;
  };
  navItems: NavItem[];
  phone: {
    number: string;
    formatted: string;
  };
  socialLinks: SocialLink[];
  contactButtonText: string;
};

// Helper function to generate a short name from the site name
const getShortName = (name: string): string => {
  // Try to extract capital letters if there are any
  const capitalLetters = name.match(/[A-Z]/g);
  if (capitalLetters && capitalLetters.length >= 2) {
    return capitalLetters.join('');
  }

  // Otherwise take the first 3 characters
  return name.substring(0, 3).toUpperCase();
};

const shortName = getShortName(siteConfig.companyName);

// Default props with values from site config
export const defaultHeaderProps: HeaderProps = {
  logo: {
    fullName: siteConfig.companyName,
    image: '/images/logo/ai-mapping-logo.png',
    textImage: '/images/logo/companyName-text.png',
  },
  // Navigation items - could be moved to site.ts config in the future for greater flexibility
  navItems: [
    { href: '/docs', label: 'Documentation' },
    // { href: '/components', label: 'Components' },
    { href: '/examples', label: 'Examples' },
    // { href: '/pricing', label: 'Pricing' },
    { href: '/blog', label: 'Blog' },
  ],
  phone: {
    number: `tel:${siteConfig.company.phone.replace(/\D/g, '')}`, // Remove non-digits
    formatted: siteConfig.company.phone,
  },
  socialLinks: [
    // Only add WhatsApp if it exists in config
    ...(siteConfig.links.whatsapp
      ? [
          {
            href: siteConfig.links.whatsapp,
            icon: <WhatsAppIcon />,
            className:
              'text-green-600 p-1 rounded hover:bg-green-100 transition',
          },
        ]
      : []),

    // Only add Telegram if it's enabled in config
    ...(siteConfig.socialMedia.telegram.enabled
      ? [
          {
            href: siteConfig.socialMedia.telegram.url,
            icon: <TelegramIcon />,
            className: 'text-blue-500 p-1 rounded hover:bg-blue-100 transition',
          },
        ]
      : []),
  ],
  contactButtonText: 'Contact Us',
};

const Header = ({
  logo = defaultHeaderProps.logo,
  navItems = defaultHeaderProps.navItems,
}: Partial<Pick<HeaderProps, 'logo' | 'navItems'>>) => {

  return (
    <header className="fixed w-full top-0 z-50 bg-background border-b border-neutral-dark">
      <div className="container mx-auto px-4 md:px-6 lg:px-8 py-3 flex items-center justify-between">
        <div className="flex items-center h-4">
          <Link
            href="/"
            className="self-start flex items-center relative -top-5 z-10 p-2"
          >
            <div className="absolute left-0 right-0 top-0 bottom-0 rounded-lg bg-black shadow-md" />
            <div className="relative">
              <Image
                src={logo.image}
                alt={logo.fullName}
                width={110}
                height={110}
                className="h-auto object-contain"
                priority
              />
              <div className="absolute inset-0 flex items-center justify-center bg-primary bg-opacity-10 text-primary font-bold text-xl rounded-lg opacity-0 fallback-logo">
                AI
              </div>
            </div>
          </Link>
          <Link href="/" className="flex items-center relative z-10 p-2">
            <div className="relative">
              <Image
                src={logo.textImage}
                alt={shortName}
                width={180}
                height={40}
                className="h-auto object-contain"
                priority
              />
              <div className="absolute inset-0 flex items-center opacity-0 fallback-logo">
                <span className="font-semibold text-xl bg-gradient-to-r from-primary via-secondary to-accent bg-clip-text text-transparent">AI-Mapping</span>
              </div>
            </div>
          </Link>

          <nav className="hidden md:flex ml-10">
            <ul className="flex space-x-6">
              {navItems.map((item, index) => (
                <li key={`nav-item-${index}`}>
                  <Link
                    href={item.href}
                    className="text-white hover:text-primary transition-colors duration-300"
                  >
                    {item.label}
                  </Link>
                </li>
              ))}
            </ul>
          </nav>
        </div>

        {/* Use the client component for interactive controls */}
        <HeaderControls />
      </div>
    </header>
  );
};

export default Header;