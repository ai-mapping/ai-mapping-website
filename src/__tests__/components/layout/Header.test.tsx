import { render, screen } from '@testing-library/react';
import Header, { defaultHeaderProps } from '@/components/layout/Header';

// Mock Next.js Image component
jest.mock('next/image', () => ({
  __esModule: true,
  default: (props: any) => {
    // eslint-disable-next-line @next/next/no-img-element
    return <img 
      src={props.src} 
      alt={props.alt} 
      width={props.width}
      height={props.height}
      className={props.className}
      data-testid={`image-${props.src.split('/').pop().split('.')[0]}`}
    />;
  },
}));

// Mock HeaderControls which is a client component
jest.mock('@/components/layout/HeaderControls', () => {
  return {
    __esModule: true,
    default: () => <div data-testid="header-controls">Header Controls</div>,
  };
});

describe('Header Component', () => {
  it('renders the header with default props', () => {
    const { container } = render(<Header />);
    
    // Check logo images are rendered
    expect(screen.getByTestId('image-ai-mapping-logo')).toBeInTheDocument();
    expect(screen.getByTestId('image-companyName-text')).toBeInTheDocument();
    
    // Check navigation items
    expect(screen.getByText('Documentation')).toBeInTheDocument();
    expect(screen.getByText('Examples')).toBeInTheDocument();
    expect(screen.getByText('Blog')).toBeInTheDocument();
    
    // Check header controls
    expect(screen.getByTestId('header-controls')).toBeInTheDocument();
    
    expect(container).toMatchSnapshot();
  });
  
  it('renders with custom logo and nav items', () => {
    const customProps = {
      logo: {
        fullName: 'Custom Company',
        image: '/custom-logo.png',
        textImage: '/custom-text.png',
      },
      navItems: [
        { href: '/custom1', label: 'Custom Link 1' },
        { href: '/custom2', label: 'Custom Link 2' },
      ],
    };
    
    render(<Header {...customProps} />);
    
    // Check custom logo images
    expect(screen.getByTestId('image-custom-logo')).toBeInTheDocument();
    expect(screen.getByTestId('image-custom-text')).toBeInTheDocument();
    
    // Check custom navigation items
    expect(screen.getByText('Custom Link 1')).toBeInTheDocument();
    expect(screen.getByText('Custom Link 2')).toBeInTheDocument();
  });
});