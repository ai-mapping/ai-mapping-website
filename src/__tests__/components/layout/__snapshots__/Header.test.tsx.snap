// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders the header with default props 1`] = `
<div>
  <header
    class="fixed w-full top-0 z-50 bg-background border-b border-neutral-dark"
  >
    <div
      class="container mx-auto px-4 md:px-6 lg:px-8 py-3 flex items-center justify-between"
    >
      <div
        class="flex items-center h-4"
      >
        <a
          class="self-start flex items-center relative -top-5 z-10 p-2"
          data-testid="next-link"
          href="/"
        >
          <div
            class="absolute left-0 right-0 top-0 bottom-0 rounded-lg bg-black shadow-md"
          />
          <div
            class="relative"
          >
            <img
              alt="AI-Mapping - Intelligent Frontend Framework"
              class="h-auto object-contain"
              data-testid="image-ai-mapping-logo"
              height="110"
              src="/images/logo/ai-mapping-logo.png"
              width="110"
            />
            <div
              class="absolute inset-0 flex items-center justify-center bg-primary bg-opacity-10 text-primary font-bold text-xl rounded-lg opacity-0 fallback-logo"
            >
              AI
            </div>
          </div>
        </a>
        <a
          class="flex items-center relative z-10 p-2"
          data-testid="next-link"
          href="/"
        >
          <div
            class="relative"
          >
            <img
              alt="AIMIFF"
              class="h-auto object-contain"
              data-testid="image-companyName-text"
              height="40"
              src="/images/logo/companyName-text.png"
              width="180"
            />
            <div
              class="absolute inset-0 flex items-center opacity-0 fallback-logo"
            >
              <span
                class="font-semibold text-xl bg-gradient-to-r from-primary via-secondary to-accent bg-clip-text text-transparent"
              >
                AI-Mapping
              </span>
            </div>
          </div>
        </a>
        <nav
          class="hidden md:flex ml-10"
        >
          <ul
            class="flex space-x-6"
          >
            <li>
              <a
                class="text-white hover:text-primary transition-colors duration-300"
                data-testid="next-link"
                href="/docs"
              >
                Documentation
              </a>
            </li>
            <li>
              <a
                class="text-white hover:text-primary transition-colors duration-300"
                data-testid="next-link"
                href="/components"
              >
                Components
              </a>
            </li>
            <li>
              <a
                class="text-white hover:text-primary transition-colors duration-300"
                data-testid="next-link"
                href="/examples"
              >
                Examples
              </a>
            </li>
            <li>
              <a
                class="text-white hover:text-primary transition-colors duration-300"
                data-testid="next-link"
                href="/pricing"
              >
                Pricing
              </a>
            </li>
            <li>
              <a
                class="text-white hover:text-primary transition-colors duration-300"
                data-testid="next-link"
                href="/blog"
              >
                Blog
              </a>
            </li>
          </ul>
        </nav>
      </div>
      <div
        data-testid="header-controls"
      >
        Header Controls
      </div>
    </div>
  </header>
</div>
`;
